wemake_python_styleguide/logic/tree/classes.py

```diff
@@ -34,3 +39,54 @@ def is_forbidden_super_class(class_name: Optional[str]) -> bool:
     if class_name in ALLOWED_BUILTIN_CLASSES:
         return False
     return is_builtin_name(class_name)
+
+
+def get_attributes(
+    node: ast.ClassDef,
+) -> Tuple[List[types.AnyAssign], List[ast.Attribute]]:
+    """Returns all class and instance attributes of a class."""
+    class_attributes = []
+    instance_attributes = []
+
+    for nd in ast.walk(node):
+        if isinstance(nd, ast.Attribute) and isinstance(nd.ctx, ast.Store):
+            instance_attributes.append(nd)
+            continue
+
+        has_assign = (
+            nodes.get_context(nd) == node and
+            getattr(nd, 'value', None)
+        )
+        if isinstance(nd, AssignNodes) and has_assign:
+            class_attributes.append(nd)
+
+    return class_attributes, instance_attributes
+
+
+def getter_setter_postfixes(node: ast.ClassDef) -> Set[str]:
+    """
+    Return postfixes of all getter or setter methods.
+
+    get_class_attribute becomes class_attribute
+
+    set_instance_attribute becomes instance_attribute
+
+    """
+    method_postfixes = set()
+    for subnode in ast.walk(node):
+        correct_context = nodes.get_context(subnode) == node
+        if isinstance(subnode, FunctionNodes) and correct_context:
+            if is_getter_or_setter(subnode):
+                method_postfixes.add(subnode.name.partition('get_')[2])
+                method_postfixes.add(subnode.name.partition('set_')[2])
+    return method_postfixes
+
+
+def is_getter_or_setter(node: types.AnyFunctionDef) -> bool:
+    """Checks if non property decorated function contains get or set prefix."""
+    if any(prefix in node.name for prefix in ('get_', 'set_')):
```

Comment:
Would this give the correct behavior for x_get_x or x_set_x?

Reply:
Will be fixed